cishouseholds/derive.py

```diff
@@ -1907,3 +1909,222 @@ def assign_regex_match_result(
         df = df.withColumn(column_name_to_assign, match_result)
 
     return df
+
+
+def get_keys_by_value(input_dict: Dict, values_to_lookup: List) -> List:
+    """
+    Returns a list of keys from the input dictionary if the dictionary's values are in the
+    given list of values.
+
+    Parameters
+    ----------
+    input_dict
+        the input dictionary
+    values_to_lookup
+        a list of values to search for in the `input_dict` and return matching keys
+
+    Raises
+    ------
+    ValueError
+        If none of the values in `values_to_lookup` are found as values of the `input_dict`.
+    """
+    result = [k for k, v in input_dict.items() if v in values_to_lookup]
+    if len(result) == 0:
+        raise ValueError("None of the values in `values_to_lookup` are found in `input_dict`")
+    return result
+
+
+def flag_records_for_generic_rules(column_to_check_in: str, categories_list: List) -> F.Column:
+    """
+    A generic function which flags records if any of the values in `column_to_check_in` match values in
+    `categories_list`.
+
+    Parameters
+    ----------
+    column_to_check_in
+        The name of the column to check
+    categories_list
+        A list of values against which values in `column_to_check_in` are checked against
+    """
+    return F.when(F.col(column_to_check_in).isin(*categories_list), F.lit(True)).otherwise(False)
+
+
+def flag_records_for_work_from_home_rules() -> F.Column:
+    """Flag records for application of "Work From Home" rules"""
+    return F.when(F.col("work_location").isNull(), F.lit(True)).otherwise(False)
```

Comment:
Similarly, can be simplified to rely on the bool type `Column` returned from `isNull`

```suggestion
    return F.col("work_location").isNull()
```